libyul/optimiser/DataFlowAnalyzer.cpp

```diff
@@ -41,6 +41,19 @@ using namespace solidity;
 using namespace solidity::util;
 using namespace solidity::yul;
 
+namespace
+{
+/// Assumes that both input vectors are sorted in ascending order
+bool hasNonemptyIntersectionSorted(std::vector<YulName> const& _vec1, std::vector<YulName> const& _vec2) {
```

Comment:
If you replaced `YulName` with a template argument, this would be a good candidate to add to other generic utilities in `libsolutil/CommonData.h`. I'd also check if we don't have one already :)

Reply:
Done:) good suggestion. I've generalized the implementation a bit as well.